### DISCUSSION TITLE
Import CSS files

### DISCUSSION THREAD

**Discussion by AnsgarH1** [shellscape/jsx-email#243] [Q&A] [answered]
Hi there, im trying to use jsx-email to generate emails in a given corporate design. The current design system is built around styled-components, but also exports css classes for legacy use cases. What is the best case to approach building templates here? I tried importing the css file provided by an internal module, but throws an error and doesn't seem to be the right approach. 

Is there a simple way to include plain css files or any styling solutions other than tailwind?

**Comment by shellscape** [comment]
Probably! I'd love to see a reproduction that showed the approach that _isn't_ working? Might help me to get a better perspective on what you're going for.

**Comment by AnsgarH1** [comment]
I built a simple demo of how i expected it to work: https://github.com/AnsgarH1/jsx-mail-css-demo/tree/main

When running a build or preview, I always get the same error:

```
TypeError: Unknown file extension ".css" for /private/var/folders/vj/kt61jh5510qdsr5x36t_qfh40000gn/T/jsx-email/build/email-LOAGP6VY.css
    at Object.getFileProtocolModuleFormat [as file:] (node:internal/modules/esm/get_format:218:9)
    at defaultGetFormat (node:internal/modules/esm/get_format:244:36)
    at defaultLoad (node:internal/modules/esm/load:122:22)
    at ModuleLoader.load (node:internal/modules/esm/loader:570:7)
    at ModuleLoader.moduleProvider (node:internal/modules/esm/loader:443:45)
    at ModuleJob._link (node:internal/modules/esm/module_job:106:19)
```

**Reply by shellscape** [reply]
Thanks, I'll play with this tonight.

**Reply by shellscape** [reply]
OK we've got a bunch of options for making this happen. Will be working on something concrete.

**Comment by shellscape** [comment]
Here's where I'm at on this, this evening. Note the `import cssText from './static/test.css'` and the results in the preview app, line 101 in the JSX, and line 11 in the HTML:

<img width="1314" alt="Screenshot 2024-11-20 at 8 14 50 PM" src="https://github.com/user-attachments/assets/79a81a88-d9de-477b-8886-2df495408106">
<img width="1314" alt="Screenshot 2024-11-20 at 8 15 04 PM" src="https://github.com/user-attachments/assets/4771ab79-25b9-4c1e-be23-7647a17162f4">
<img width="1314" alt="Screenshot 2024-11-20 at 8 15 08 PM" src="https://github.com/user-attachments/assets/7a2a8f36-5cd7-474f-898b-df9bd75e67b3">

**Reply by shellscape** [reply]
I'm investigating returning a component from an import as well. the TypeScript typing is a bit tricky

**Reply by shellscape** [reply]
For now, https://github.com/shellscape/jsx-email/pull/244

**Comment by shellscape** [comment]
This has been merged, docs have been updated, and will be available in a few minutes.

**Comment by AnsgarH1** [comment, marked as answer]
Hey sorry, for the late reply. Thanks for the implementation, i could build a small proof of concept for our client! 

Happy holidays!
